roles/nova-compute/tasks/main.yml

```diff
@@ -103,9 +120,77 @@
             dest=/etc/neutron/neutron.conf
             mode=0640 owner=root group=neutron
   notify: restart neutron openvswitch agent
-  when: use_neutron
+  when: use_neutron and (openstack_release == 'icehouse' or openstack_release == 'havana')
   tags: nova
 
+- name: Update neutron.conf file for Juno
+  command: openstack-config --verbose --set /etc/neutron/neutron.conf {{ item }}
+  with_items:
+    - "DEFAULT auth_strategy keystone"
+    - "keystone_authtoken auth_host {{ keystone_admin_vip | default('127.0.0.1') }}"
+    - "keystone_authtoken auth_port {{ keystone_auth_port | default(35357) }}"
+    - "keystone_authtoken auth_protocol {{ keystone_auth_protocol | default('http') }}"
+    - "keystone_authtoken admin_tenant_name services"
+    - "keystone_authtoken admin_user neutron"
+    - "keystone_authtoken admin_password {{ neutron_pass }}"
+    - "DEFAULT rabbit_hosts {{ rabbit_hosts }}"
+    - "DEFAULT rabbit_ha_queues true"
+    - "DEFAULT notification_driver neutron.openstack.common.notifier.rpc_notifier"
+  register: cmd
+  changed_when: "'unchanged' not in cmd.stderr"
+  when: openstack_release == 'juno'
+
+- name: Update ovs_neutron_plugin.ini file for Juno
+  command: openstack-config --verbose --set /etc/neutron//plugins/openvswitch/ovs_neutron_plugin.ini {{ item }}
+  with_items:
+    - "agent tunnel_types vxlan"
```

Comment:
line 146 - 155 most of this parameters have available variables to use.  See `ovs_neutron_plugini.j2`